### PR TITLE
Pylnk: Init at 0.4.2

### PR DESCRIPTION
**Pull request by FedX-sudo**
###### Description of changes
Added pylnk to Nixpkgs because it is required for CrackMapExec.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pylnk3/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+, fetchPypi
+, pytest
+, twine
+, invoke
+}:
+
+buildPythonPackage rec {
+  pname = "pylnk3";
+  version = "0.4.2";
+
```
Comment (fabaff) [inline pkgs/development/python-modules/pylnk3/default.nix:12]: No support for [Python < 3.6](https://github.com/strayge/pylnk/blob/master/setup.py#L32).

```suggestion

  disabled = pythonOlder "3.6";
```

File: pkgs/development/python-modules/pylnk3/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchPypi {
+    inherit version;
+    pname = "pylnk3";
+    sha256 = "sha256-yu4BNvYai3iBVNyOfAOsLd5XrcFw8cR4arRyFJHKbpk=";
+  };
+
+   propagatedBuildInputs = [
+     pytest
```
Comment (fabaff) [inline pkgs/development/python-modules/pylnk3/default.nix:20]: Please use `pytestCheckHook` in `checkInputs` for the test requirements.

```bash
  checkInputs = [
    pytestCheckHook
  ];
```

File: pkgs/development/python-modules/pylnk3/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+     twine
```
Comment (fabaff) [inline pkgs/development/python-modules/pylnk3/default.nix:21]: `twine` is usually not a runtime requirement.

File: pkgs/development/python-modules/pylnk3/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+   ];
+
+  pythonImportsCheck = [
+    "pylnk3"
+  ];
+
+  meta = with lib; {
+    description = "Python library for reading and writing Windows shortcut files (.lnk)";
+    homepage = "https://github.com/strayge/pylnk";
+    license = with licenses; [ lgpl3 ];
```
Comment (fabaff) [inline pkgs/development/python-modules/pylnk3/default.nix:32]: I can't find any mentioning that later releases of the LGPL are supported.

```suggestion
    license = with licenses; [ lgpl3Only ];
```

File: pkgs/top-level/python-packages.nix [hunk 5]
```diff
@@ -186,6 +186,8 @@ in {
 
   actdiag = callPackage ../development/python-modules/actdiag { };
 
+  adafruit-io = callPackage ../development/python-modules/adafruit-io { };
```
Comment (fabaff) [inline pkgs/top-level/python-packages.nix:189]: Looks like unrelated.

Comment (FedX-sudo) [reply pkgs/top-level/python-packages.nix:189]: It is, I was getting a merge conflict error, and added that fixed it...

Comment (fabaff) [reply pkgs/top-level/python-packages.nix:189]: `adafruit-io` was added approx. a month ago. Did you try to rebase your branch on the latest master?

Comment (FedX-sudo) [reply pkgs/top-level/python-packages.nix:189]: No. I will try that.

**Comment by FedX-sudo** [comment]
Fixed it! Dang Git can be annoying.

File: pkgs/top-level/python-packages.nix [hunk 6]
```diff
@@ -7122,6 +7122,8 @@ in {
 
   py-lru-cache = callPackage ../development/python-modules/py-lru-cache { };
 
+  pylnk3 = callPackage ../development/python-modules/pylnk3/default.nix
```
Comment (fabaff) [inline pkgs/top-level/python-packages.nix:7125]:
```suggestion
  pylnk3 = callPackage ../development/python-modules/pylnk3 { };
```

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/166864#issuecomment-1087279364)
- [ ] all depending packages build
- [x] tested on x86_64

Looks good to me.
